doc/tut2.rst

```diff
@@ -603,19 +603,22 @@ Turning the ``log`` proc into a template solves this problem:
     x = 4
   log("x has the value: " & $x)
 
-The parameters' types can be ordinary types or the meta types ``expr``
-(stands for *expression*), ``stmt`` (stands for *statement*) or ``typedesc``
-(stands for *type description*). If the template has no explicit return type,
-``stmt`` is used for consistency with procs and methods.
+The parameters' types can be ordinary types or the meta types ``untyped``,
+``typed``, or ``typedesc``.
+``typedesc`` stands for *type description*, and ``untyped`` means symbol lookups and
+type resolution is not performed before the expression is passed to the template.
 
-If there is a ``stmt`` parameter it should be the last in the template
-declaration. The reason is that statements can be passed to a template
+If the template has no explicit return type,
+``void`` is used for consistency with procs and methods.
+
+If there is a ``typed`` parameter, it should be the last in the template
```

Comment:
This sentence makes no sense anymore. Make this "To pass a block of statements to the template, use 'untyped' for the last parameter:"